modules/uploads/utils/types.ts

```diff
@@ -0,0 +1,75 @@
+import {
+	Files as PrismaFiles,
+	Upload as _PrismaUpload,
+} from "../_gen/prisma/default.d.ts";
+
+interface PrismaUpload extends Omit<_PrismaUpload, "deletedAt"> {
+	files: PrismaFiles[];
+}
+
+export interface Upload {
+	id: string;
+	userId: string | null;
+
+	bucket: string;
+	contentLength: string;
+
+	files: UploadFile[];
+
+	createdAt: string;
+	updatedAt: string;
+	completedAt: string | null;
+}
+
+export interface UploadFile {
+	path: string;
+	mime: string | null;
+	contentLength: string;
+}
+
+export interface PresignedUpload extends Omit<Upload, "files"> {
+	files: PresignedUploadFile[];
+}
+
+export interface PresignedUploadFile extends UploadFile {
+	presignedUrls: PresignedUrl[];
+}
+
+export interface PresignedUrl {
+	url: string;
+
+	partNumber: number;
+	contentLength: string;
+	offset: string;
+}
+
+export type PrismaUploadWithOptionalFiles = Omit<PrismaUpload, "files"> & Partial<Pick<PrismaUpload, "files">>;
+export type UploadWithOptionalFiles = Omit<Upload, "files"> & Partial<Pick<Upload, "files">>;
+
+
+type InputType<HasFiles extends boolean> = HasFiles extends true ? PrismaUpload : PrismaUploadWithOptionalFiles;
+type OutputType<HasFiles extends boolean> = HasFiles extends true ? Upload : UploadWithOptionalFiles;
+export function prismaToOutput<HasFiles extends boolean>(upload: InputType<HasFiles>, hasFiles: HasFiles): OutputType<HasFiles> {
```

Comment:
Can we simplfying this to not use TypeScript magic? A big goal of OpenGB is legibility to people used to C# & simpler languages.